File: src/mysqlcookbook/recipes/api/01_connect/connect2.rb
#!/usr/bin/ruby
# connect2.rb: connect to the MySQL server, showing how to specify
# a port number or Unix domain socket path explicitly.

require "mysql2"

# Set the port explicitly

begin
#@ _FRAG_PORT_
  puts "Connect using port"
  client = Mysql2::Client.new(:host => "127.0.0.1",
                              :port => 3307,
                              :username => "cbuser",
                              :password => "cbpass",
                              :database => "cookbook")
#@ _FRAG_PORT_
  puts "Connected"
rescue
  puts "Cannot connect to server"
else
  client.close()
  puts "Disconnected"
end

# Set the socket file explicitly

begin
  puts "Connect using socket"
#@ _FRAG_SOCKET_
  client = Mysql2::Client.new(:host => "localhost",
                              :port => "/tmp/mysql.sock",
                              :username => "cbuser",
                              :password => "cbpass",
                              :database => "cookbook")
#@ _FRAG_SOCKET_
  puts "Connected"
rescue
  puts "Cannot connect to server"
else
  client.close()
  puts "Disconnected"
end

File: src/mysqlcookbook/recipes/api/01_connect/connect2.py
#!/usr/bin/python3
# connect2.py: show other ways to connect to the MySQL server

# * Use a dictionary to specify connection arguments
# * How to specify a port number or Unix domain socket path explicitly.

import mysql.connector

# Use a dictionary to specify connection arguments

try:
#@ _FRAG_DICTIONARY_
  conn_params = {
    "database": "cookbook",
    "host": "localhost",
    "user": "cbuser",
    "password": "cbpass",
  }
  conn = mysql.connector.connect(**conn_params)
  print("Connected")
#@ _FRAG_DICTIONARY_
except:
  print("Cannot connect to server")
else:
  conn.close()
  print("Disconnected")

# Set the port explicitly

try:
#@ _FRAG_PORT_
  conn_params = {
    "database": "cookbook",
    "host": "127.0.0.1",
    "port": 3307,
    "user": "cbuser",
    "password": "cbpass",
  }
  conn = mysql.connector.connect(**conn_params)
#@ _FRAG_PORT_
except:
  print("Cannot connect to server")
else:
  print("Connected")
  conn.close()
  print("Disconnected")

# Set the socket file explicitly

try:
#@ _FRAG_SOCKET_
  conn_params = {
    "database": "cookbook",
    "unix_socket": "/var/tmp/mysql.sock",
    "user": "cbuser",
    "password": "cbpass",
  }
  conn = mysql.connector.connect(**conn_params)
  print("Connected")
#@ _FRAG_SOCKET_
except:
  print("Cannot connect to server")
else:
  conn.close()
  print("Disconnected")
